Fix bug where Cilium incorrectly configures externalTrafficPolicy for ClusterIP services in Gateway API

syedazeez337 opened the pull request:
## Description

This PR fixes an issue where the gateway controller was setting `externalTrafficPolicy` for ClusterIP services, which is invalid according to Kubernetes validation. The fix ensures that `externalTrafficPolicy` is only set for LoadBalancer and NodePort service types.

## How has this been tested?

- Added unit tests for the `toExternalTrafficPolicy` function to verify it returns an empty string for ClusterIP services
- Added a test case for a ClusterIP service to ensure the fix works correctly
- Manually tested with a Kind cluster by creating a Gateway with a `CiliumGatewayClassConfig` that specifies `spec.service.type: ClusterIP`

## Related issues

Fixes: #38852

## Checklist

- [x] I have read the [Cilium contributing guide](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/)
- [x] I have added unit tests to verify the fix
- [x] I have tested the changes on a Kind cluster
- [x] I have updated the documentation if necessary

qmonnet reviewed (approved):
Not sure what got `docs-structure` to be pulled for review, I guess this was from a previous version of the PR. Looks good as far as I'm concerned. Thanks!

sayboras commented on operator/pkg/model/translation/gateway-api/translator_test.go:
```diff
@@ -35,6 +35,7 @@ func Test_translator_Translate(t *testing.T) {
 		{name: "basic_http_listener_nodeport"},
 		{name: "basic_http_listener_external_traffic_policy"},
 		{name: "basic_http_listener_load_balancer"},
+		{name: "basic_http_listener_clusterip"},
```
seems like this is the only change, and the test is passing :thinking:. 

Do we need to change anything in translator.go ?

syedazeez337 committed:
gateway-api: Fix externalTrafficPolicy for ClusterIP services

Fixes issue where Cilium incorrectly sets the externalTrafficPolicy field
for ClusterIP services in Gateway API, causing service creation to fail
with the error: "Service is invalid: spec.externalTrafficPolicy: Invalid
value: 'Cluster': may only be set for externally-accessible services".

This fix modifies the toExternalTrafficPolicy function to check the service
type and only set the externalTrafficPolicy field for LoadBalancer and
NodePort service types, returning an empty string for ClusterIP services.

Fixes: #38852

Signed-off-by: Syed Azeez <syedazeez337@gmail.com>
